src/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public partial class JsonContent : HttpContent
+    {
+        internal const string JsonMediaType = "application/json";
+        internal static readonly MediaTypeHeaderValue DefaultMediaType = MediaTypeHeaderValue.Parse(string.Format("{0} {1}", JsonMediaType, Encoding.UTF8.WebName));
+
+        private readonly JsonSerializerOptions? _jsonSerializerOptions;
+        public Type ObjectType { get; }
+        public object? Value { get; }
+
+        private JsonContent(object? value, Type inputType, MediaTypeHeaderValue mediaType, JsonSerializerOptions? options)
+        {
+            if (mediaType == null)
```

Comment:
I thought we were allowing mediaType to be null, and actually making it an optional parameter (defaulting to null) to avoid needing additional Create overloads that didn't take it.

Reply:
In that case we need to mark it as such (`MediaTypeHeaderValue?`).

Reply:
Yes, that was the decision in API review as far as I remember: `MediaTypeHeaderValue? mediaType = null`.